pkg/logutil/zap_text_encoder.go

```diff
@@ -0,0 +1,610 @@
+}
+
+func (enc *textEncoder) AppendUint64(val uint64) {
+	enc.addElementSeparator()
+	enc.buf.AppendUint(val)
+}
+
+func (enc *textEncoder) AddComplex64(k string, v complex64) { enc.AddComplex128(k, complex128(v)) }
+func (enc *textEncoder) AddFloat32(k string, v float32)     { enc.AddFloat64(k, float64(v)) }
+func (enc *textEncoder) AddInt(k string, v int)             { enc.AddInt64(k, int64(v)) }
+func (enc *textEncoder) AddInt32(k string, v int32)         { enc.AddInt64(k, int64(v)) }
+func (enc *textEncoder) AddInt16(k string, v int16)         { enc.AddInt64(k, int64(v)) }
+func (enc *textEncoder) AddInt8(k string, v int8)           { enc.AddInt64(k, int64(v)) }
+func (enc *textEncoder) AddUint(k string, v uint)           { enc.AddUint64(k, uint64(v)) }
+func (enc *textEncoder) AddUint32(k string, v uint32)       { enc.AddUint64(k, uint64(v)) }
+func (enc *textEncoder) AddUint16(k string, v uint16)       { enc.AddUint64(k, uint64(v)) }
+func (enc *textEncoder) AddUint8(k string, v uint8)         { enc.AddUint64(k, uint64(v)) }
+func (enc *textEncoder) AddUintptr(k string, v uintptr)     { enc.AddUint64(k, uint64(v)) }
+func (enc *textEncoder) AppendComplex64(v complex64)        { enc.AppendComplex128(complex128(v)) }
+func (enc *textEncoder) AppendFloat64(v float64)            { enc.appendFloat(v, 64) }
+func (enc *textEncoder) AppendFloat32(v float32)            { enc.appendFloat(float64(v), 32) }
+func (enc *textEncoder) AppendInt(v int)                    { enc.AppendInt64(int64(v)) }
+func (enc *textEncoder) AppendInt32(v int32)                { enc.AppendInt64(int64(v)) }
+func (enc *textEncoder) AppendInt16(v int16)                { enc.AppendInt64(int64(v)) }
+func (enc *textEncoder) AppendInt8(v int8)                  { enc.AppendInt64(int64(v)) }
+func (enc *textEncoder) AppendUint(v uint)                  { enc.AppendUint64(uint64(v)) }
+func (enc *textEncoder) AppendUint32(v uint32)              { enc.AppendUint64(uint64(v)) }
+func (enc *textEncoder) AppendUint16(v uint16)              { enc.AppendUint64(uint64(v)) }
+func (enc *textEncoder) AppendUint8(v uint8)                { enc.AppendUint64(uint64(v)) }
+func (enc *textEncoder) AppendUintptr(v uintptr)            { enc.AppendUint64(uint64(v)) }
+
+func (enc *textEncoder) Clone() zapcore.Encoder {
+	clone := enc.cloned()
+	clone.buf.Write(enc.buf.Bytes())
+	return clone
+}
+
+func (enc *textEncoder) cloned() *textEncoder {
+	clone := getTextEncoder()
+	clone.EncoderConfig = enc.EncoderConfig
+	clone.spaced = enc.spaced
+	clone.openNamespaces = enc.openNamespaces
+	clone.buf = _pool.Get()
+	return clone
+}
+
+func (enc *textEncoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
+	final := enc.cloned()
+	if final.TimeKey != "" {
+		final.beginQuoteFiled()
+		final.AppendTime(ent.Time)
+		final.endQuoteFiled()
+	}
+
+	if final.LevelKey != "" {
+		final.beginQuoteFiled()
+		cur := final.buf.Len()
+		final.EncodeLevel(ent.Level, final)
```

Comment:
I'm not sure about all levels. Remember that in RFC there can be only FATAL, ERROR, WARN, INFO and DEBUG